com.minres.coredsl/src/com/minres/coredsl/analysis/CoreDslAnalyzer.xtend

```diff
@@ -625,6 +627,25 @@ class CoreDslAnalyzer {
 					ctx.acceptError("Cannot implicitly convert " + valueType + " to " + type, declarator,
 						CoreDslPackage.Literals.DECLARATOR__TEQUALS, -1, IssueCodes.InvalidAssignmentType);
 				}
+			} else if(type.isArrayType) {
+				val listInitializer = initializer as ListInitializer;
+				val arrayType = type as ArrayType;
+				if (arrayType.count != listInitializer.initializers.size)
+					ctx.acceptError("List initializer size does not match array size", declarator,
+						CoreDslPackage.Literals.DECLARATOR__TEQUALS, -1, IssueCodes.InvalidAssignmentType);
+				for (subInitializer : listInitializer.initializers) {
+					if(subInitializer instanceof ExpressionInitializer) {
+						val expressionInitializer = subInitializer as ExpressionInitializer;
```

Comment:
This cast is unnecessary. In the scope of `if(subInitializer instanceof ExpressionInitializer)` you can use `subInitializer` as if it was declared as `ExpressionInitializer` (one of the comfort features of Xtend).

Reply:
🤯